### ISSUE TITLE
Dots in package names are getting replaced by underscores since 1.2.0

### ISSUE THREAD

**Issue by SvenBecker** [python-poetry/poetry#6590]
- **Poetry version**: 1.2.0/1.2.1
- **Python version**: 3.8/3.9
- **OS version and name**: macOS/Linux

- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a recommended method.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below.

## Issue

Given a `pyproject.toml` file like the following

```toml
[tool.poetry]
name = "com.namespace.package"
version = "5.3.0"
```

previously the resulting wheel name after running `poetry build` was `com.namespace.package-5.3.0-py3-none-any.whl`. Now it is
`com_namespace_package-5.3.0-py3-none-any.whl`. Same applies to metadata on various locations. Unfortunately we can not simply rename our packages due to permission handling based on artifact names.

**Comment by rkhwaja** [comment]
This is still broken in 1.3.2.

My package name is fs.onedrivefs. The files produced by poetry build are fs_onedrivefs*. The dist-info folder in the wheel is similarly renamed. The name in the METADATA file is changed to fs-onedrivefs

**Comment by neersighted** [comment]
That's not a bug, that's the [expected filename](https://peps.python.org/pep-0491/#escaping-and-unicode) as mandate by the packaging standards.

**Comment by rkhwaja** [comment]
I don't think that covers the changing of the name field in the METADATA file. It causes the name of the package to be wrong in `pip freeze` for instance.

**Comment by petergaultney** [comment]
PEP 491 is apparently deferred - are we certain it's a packaging standard?

Incidentally, `importlib.metadata.version` does not successfully read the METADATA for packages published by `poetry`, because of the dots being replaced by underscores.

**Comment by neersighted** [comment]
Sorry, that was a bad citation. [PEP 625](https://peps.python.org/pep-0625/) is the authority for sdist names, and [PEP 427](https://peps.python.org/pep-0427/#file-name-convention) is canonical for wheels. Both impose the same escaping rules.
